LowPoleMassPrecision = {Hm} in CPV model cause a crash

Following change cause error
```
DefaultPoleMassPrecision = HighPrecision;
HighPoleMassPrecision    = {hh, Ah};
MediumPoleMassPrecision  = {};
LowPoleMassPrecision     = {Hpm};
```
This is specific only to CPV models. Was discovered by Vytautas in his custom model but happens also in CMSSMCPV. Unfortunately I'm not familiar with this part of the code. Anyone has any idea?
```
In file included from models/CMSSMCPV/CMSSMCPV_mass_eigenstates.cpp:34:
src/eigen_utils.hpp:248:4: error: no matching function for call to 'reorder_vector'
   reorder_vector(v, matrix.diagonal().array().eval());
   ^~~~~~~~~~~~~~
models/CMSSMCPV/CMSSMCPV_mass_eigenstates.cpp:8662:4: note: in instantiation of function template specialization 'flexiblesusy::reorder_vector<Eigen::Matrix<std::complex<double>, 2, 2, 0, 2, 2>>' requested here
   reorder_vector(MHpm_reordered, get_mass_matrix_Hpm());
   ^
src/eigen_utils.hpp:222:6: note: candidate template ignored: deduced conflicting types for parameter 'Real' ('double' vs. 'std::complex<double>')
void reorder_vector(
     ^
src/eigen_utils.hpp:244:6: note: candidate template ignored: could not match 'MatrixBase' against 'Array'
void reorder_vector(
     ^
1 error generated.
```

--- a/src/eigen_utils.hpp
+++ b/src/eigen_utils.hpp
@@ -236,7 +236,9 @@ void reorder_vector(
 }
 
 /**
- * @brief reorders vector v according to ordering of diagonal elements in mass_matrix
+ * @brief reorders vector v according to ordering of the magitude of
+ * the diagonal elements in matrix
+ *
  * @param v vector with elementes to be reordered
  * @param matrix matrix with diagonal elements with reference ordering
  */
@@ -245,7 +247,7 @@ void reorder_vector(
    Eigen::Array<double,Eigen::MatrixBase<Derived>::RowsAtCompileTime,1>& v,
    const Eigen::MatrixBase<Derived>& matrix)
 {
-   reorder_vector(v, matrix.diagonal().array().eval());
+   reorder_vector(v, matrix.diagonal().array().cwiseAbs().eval());
 }
 
 /// sorts an Eigen array

--- a/test/test_eigen_utils.cpp
+++ b/test/test_eigen_utils.cpp
@@ -163,6 +163,40 @@ BOOST_AUTO_TEST_CASE(test_reorder_vector_with_matrix)
    BOOST_CHECK_EQUAL(vec1(2), 2);
 }
 
+BOOST_AUTO_TEST_CASE(test_reorder_vector_with_matrix_abs)
+{
+   Eigen::Array<double,3,1> vec1;
+   vec1 << 1, 2, 3;
+
+   Eigen::Matrix<double,3,3> matrix;
+   matrix << 3, 0, 0,
+             0, 1, 0,
+             0, 0, -2; // uses abs() of diagonal values
+
+   reorder_vector(vec1, matrix);
+
+   BOOST_CHECK_EQUAL(vec1(0), 3);
+   BOOST_CHECK_EQUAL(vec1(1), 1);
+   BOOST_CHECK_EQUAL(vec1(2), 2);
+}
+
+BOOST_AUTO_TEST_CASE(test_reorder_vector_with_matrix_complex)
+{
+   Eigen::Array<double,3,1> vec1;
+   vec1 << 1, 2, 3;
+
+   Eigen::Matrix<std::complex<double>,3,3> matrix;
+   matrix << 3, 0, 0,
+             0, 1, 0,
+             0, 0, 2;
+
+   reorder_vector(vec1, matrix);
+
+   BOOST_CHECK_EQUAL(vec1(0), 3);
+   BOOST_CHECK_EQUAL(vec1(1), 1);
+   BOOST_CHECK_EQUAL(vec1(2), 2);
+}
+
 BOOST_AUTO_TEST_CASE(test_remove_if_equal)
 {
    Eigen::Array<double,5,1> src;

